_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+const CognitoUser = AmazonCognitoIdentity.CognitoUser;
+const CognitoUserPool = AmazonCognitoIdentity.CognitoUserPool;
+const AuthenticationDetails = AmazonCognitoIdentity.AuthenticationDetails;
+
+/*
+ * from /sorgerlab/minerva-client-js/master/index.js
+ */
+
+const authenticateUser = function(cognitoUser, authenticationDetails) {
+  return new Promise(function(resolve, reject) {
+    cognitoUser.authenticateUser(authenticationDetails, {
+      onSuccess: result => resolve(result),
+      onFailure: err => reject(err),
+      mfaRequired: codeDeliveryDetails => reject(codeDeliveryDetails),
+      newPasswordRequired: (fields, required) => reject({fields, required})
+    });
+  });
+};
+
+const authenticate = function(username, pass) {
+
+  return pass.then(function(password) {
+
+    const minervaPoolId = 'us-east-1_YuTF9ST4J'; 
+    const minervaClientId = '6ctsnjjglmtna2q5fgtrjug47k';
+    const minervaPool = new CognitoUserPool({
+      UserPoolId : minervaPoolId,
+      ClientId : minervaClientId
+    });
+ 
+    const cognitoUser = new CognitoUser({
+      Username: username,
+      Pool: minervaPool
+    });
+ 
+    const authenticationDetails = new AuthenticationDetails({
+      Username: username,
+      Password: password
+    });
+ 
+    return authenticateUser(cognitoUser, authenticationDetails)
+      .then(response => response.getIdToken().getJwtToken());
+  });
+};
+
+
+const flatten = function(items) {
+  return items.reduce(function(flat, item) {
+    return flat.concat(item);
+  });
+};
+
+const round4 = function(n) {
+  const N = Math.pow(10, 4);
+  return Math.round(n * N) / N;
+};
+const show5 = function(n) {
+  return ('' + n).slice(0, 5);
+};
+
+const modulo = function(i, n) {
+  return ((i % n) + n) % n;
+};
+
+const encode = function(txt) {
+  return btoa(encodeURIComponent(txt));
+};
+
+const decode = function(txt) {
+  try {
+    return decodeURIComponent(atob(txt));
+  }
+  catch (e) {
+    return '';
+  }
+};
+
+const arrayEqual = function(a, b) {
+  if (a.length != b.length) {
+    return false;
+  }
+  var pairs = [];
+  for (var i = 0; i < a.length; i++) {
+    pairs.push([a[i], b[i]]);
+  }
+  return pairs.every(function(p) {
+    return p[0] == p[1];
+  });
+};
+
+const dFromWaypoint = function(waypoint) {
+  return encode(waypoint.Description);
+};
+
+const nFromWaypoint = function(waypoint) {
+  return encode(waypoint.Name);
+};
+
+const gFromWaypoint = function(waypoint, cgs) {
+  const cg_name = waypoint.Group;
+  return index_name(cgs, cg_name);
+};
+const vFromWaypoint = function(waypoint) {
+  return [
+    waypoint.Zoom,
+    waypoint.Pan[0],
+    waypoint.Pan[1],
+  ];
+};
+
+const oFromWaypoint = function(waypoint) {
+  return [
+    waypoint.Overlay.x,
+    waypoint.Overlay.y,
+    waypoint.Overlay.width,
+    waypoint.Overlay.height,
+  ];
+};
+
```

Comment:
dFromWaypoint, nFromWaypoint, gFromWaypoint, vFromWaypoint, oFromWaypoint

all transform configuration attributes for updating the application state.